### PR TITLE
Vdb numpy bug

### PR DESCRIPTION
**Pull request by maxm17**
Could someone please check my tests as I've not worked with byte arrays before

### CODE REVIEW

**Review by cflynn3 — APPROVED** [review]
This looks good to me, but let's wait for @andy-beer to review it in case there's anything I've missed.

File: tests/unit_tests/olio/test_vdb.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import pytest
+
+import resqpy.olio.vdb as vdb
+import numpy as np
+
+
+@pytest.mark.parametrize("byte_array, expected_array", [(b" \x01\x00\x00\xae\x01\x00\x00", np.array([288, 430])),
```
Comment (andy-beer) [inline tests/unit_tests/olio/test_vdb.py:7]: I suggest replacing the leading space in the first byte array with the hexadecimal form, \x20 I believe.

File: resqpy/olio/vdb.py [hunk 2]
```diff
@@ -221,8 +221,8 @@ def __init__(self, fp, place, item_type, count, max_count):
             self.a = np.empty((count, 2), dtype = int)
             self.a[:, :] = nda[:, :]
         else:
-            self.a = np.empty((count,), dtype = dtype)
-            self.a.data = fp.read(count * byte_size)  # todo: check C ordering; endianess etc.
+            b = fp.read(count * byte_size)
+            self.a = np.frombuffer(buffer = b, dtype = dtype, count = count)
```
Comment (andy-beer) [inline resqpy/olio/vdb.py:225]: Does frombuffer() have an explicit byte size per element, or is it deduced from count (and the size of buffer), or does dtype determine the size per element?  Vdb is using 32bit floats and ints.

**Review by andy-beer — COMMENTED** [review]
Looks good.  I'd like to see one or two of the bp notebooks tested with the new code.

File: tests/unit_tests/olio/test_vdb.py (hunk 1 above)
Comment (maxm17) [reply tests/unit_tests/olio/test_vdb.py]: Added

File: resqpy/olio/vdb.py (hunk 2 above)
Comment (maxm17) [reply resqpy/olio/vdb.py:225]: It is deduced from count [https://numpy.org/doc/stable/reference/generated/numpy.frombuffer.html](url)

**Comment by maxm17** [comment]
@andy-beer Are there any specific notebooks I should use?

**Comment by maxm17** [comment]
Tested on the wren data vdb files, which produces the same output

**Pull request merged**